core/src/main/java/org/apache/calcite/rel/RelDistributions.java

```diff
@@ -80,6 +80,15 @@ public static RelDistribution range(Collection<? extends Number> numbers) {
     return RelDistributionTraitDef.INSTANCE.canonize(trait);
   }
 
+  /** Creates a distribution. */
+  public static RelDistribution distribution(
+      RelDistribution.Type type, List<Number> keys) {
+    ImmutableIntList list = ImmutableIntList.copyOf(keys);
+    RelDistributionImpl trait =
+        new RelDistributionImpl(type, list);
+    return RelDistributionTraitDef.INSTANCE.canonize(trait);
+  }
+
```

Comment:
Can you explain the reason that you add this method? Besides, it might be better if the type of `keys` is `Collection<? extends Number>`.

Reply:
+1 for `Collection<? extends Number>`

Reply:
Because `RelDistributionImpl` is private, need this method to create a RelDistributionImpl object from json string.